src/sage/combinat/colored_permutations.py

```diff
@@ -1235,19 +1247,60 @@ def to_cycles(self, singletons=True, use_min=True, negative_singletons=True):
             cycle = [cycle_first]
             l[i], next_val = False, l[i]
             s = self._colors[i]
+            add_neg = True
             while next_val != cycle_first:
                 cycle.append(s * next_val)
                 s *= self._colors[next_val - 1]
                 l[next_val - 1], next_val = False, l[next_val - 1]
             if s != 1:
                 cycle.extend([-e for e in cycle])
+                add_neg = False
 
             # Add the cycle to the list of cycles
             if singletons or len(cycle) > 1:
                 cycles.append(tuple(cycle))
+                if negative_cycles and add_neg:
+                    cycles.append(tuple([-e for e in cycle]))
 
         return cycles
 
+    def cycle_type(self):
+        r"""
+        Return a pair of partitions of ``len(self)`` corresponding to the
+        signed cycle type of ``self``.
+
+        A *negative cycle* is a cycle `C = (c_0, \ldots, c_{2k-1})` such that
+        `c_0 = c_k`. Any other cycle is positive. For any cycle `C`, we
+        ignore the cycle `-C` (provided this is a different cycle).
+
+        EXAMPLES::
+
+            sage: pi = SignedPermutations(7)([2,-1,4,-6,-5,-3,7])
```

Comment:
This should be simplified to `SignedPermutation([2,-1,4...])`.

Reply:
I extracted the group to its own line.